evm_loader/program/src/precompile_contracts.rs

```diff
@@ -307,6 +312,83 @@ pub fn erc20_wrapper<'a, B: AccountStorage>(
     }
 }
 
+// Neon token method ids:
+//--------------------------------------------------
+// withdraw(bytes32)           => 8e19899e
+//--------------------------------------------------
+const NEON_TOKEN_METHOD_WITHDRAW_ID: &[u8; 4]       = &[0x8e, 0x19, 0x89, 0x9e];
+const GALAN: u64 = 1_000_000_000;
+
+/// Call inner `neon_token`
+#[must_use]
+#[allow(clippy::too_many_lines)]
+pub fn neon_token<'a, B: AccountStorage>(
+    input: &[u8],
+    context: &evm::Context,
+    state: &mut ExecutorState<'a, B>
+)
+    -> Capture<(ExitReason, Vec<u8>), Infallible>
+{
+    debug_print!("neon_token({})", hex::encode(&input));
+
+    let (method_id, rest) = input.split_at(4);
+    let method_id: &[u8; 4] = method_id.try_into().unwrap_or(&[0_u8; 4]);
+
+    if method_id == NEON_TOKEN_METHOD_WITHDRAW_ID  {
+        if state.metadata().is_static() {
+            let revert_message = b"neon_token: withdraw is not allowed in static context".to_vec();
+            return Capture::Exit((ExitReason::Revert(evm::ExitRevert::Reverted), revert_message))
+        }
+
+        let source = context.address; // caller contract
+        let amount_alan = context.apparent_value;
+
+        // owner of the associated token account
+        let destination = array_ref![rest, 0, 32];
+        let destination = Pubkey::new_from_array(*destination);
+
+        if amount_alan % GALAN != U256::from(0) {
+            if let Some(error) = revert_transfer(context, state) {
+                return Capture::Exit(error);
+            }
+            let revert_message = b"neon_token: amount must be divisible by 1 billion".to_vec();
+            return Capture::Exit((ExitReason::Revert(evm::ExitRevert::Reverted), revert_message))
+        }
+
+        let amount_giga_alan: U256 = amount_alan / GALAN;
```

Comment:
amount_alan % GALAN on line 350
amount_alan / GALAN on line 358
It is more efficient to use [div_mod](https://docs.rs/primitive-types/latest/primitive_types/struct.U256.html#method.div_mod) to compute both of them at the same time

Reply:
Fixed